src/Features/Core/Portable/MoveStaticMembers/MoveStaticMembersWithDialogCodeAction.cs

```diff
@@ -147,5 +154,106 @@ private static TypeKind GetNewTypeKind(ImmutableArray<ITypeParameterSymbol> type
 
             return docEditor.GetChangedDocument();
         }
+
+        private static async Task<Solution> RefactorReferencesAsync(
+            ILookup<DocumentId, (ReferenceLocation location, bool isExtension)> locationsToDoc,
+            Solution solution,
+            INamedTypeSymbol newType,
+            CancellationToken cancellationToken)
+        {
+            var solutionEditor = new SolutionEditor(solution);
+            foreach (var docGroup in locationsToDoc)
```

Comment:
this worries me about how you can be jumping randomly across documents.  ideally we would group documents by project so we can get the compilation for a project once, process the entire project, then throw it away and move to the next project.